.github/workflows/cleanup-old-workflows.yml

```diff
@@ -0,0 +1,26 @@
+# Copyright 2022 The Magma Authors.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: "Cleanup Github Workflows"
+
+on:
+  schedule:
+    - cron: '42 * * * *'
+
+jobs:
+  cleanup-workflows:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: Mattraks/delete-workflow-runs@c1a2f0de3fa4134a2357de83c92acff6e298bb8a # pin@v2.0.3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          retain_days: 90
```

Comment:
While I like 90 days better than 30 days, I would still prefer to keep workflow runs for a longer time, like a year. I think we should also raise that question in Slack as it's strictly speaking a breaking change, and we shouldn't surprise people with this. I assume not everybody with a stake in this is looking at this review.

Reply:
A public post in Slack was made: https://magmacore.slack.com/archives/C018J8UMGMR/p1661418380688709.

Reply:
I also think that it would be nice to be able to analyse how build times etc. changed within the last year - at least for the main workflows, incl. `bazel.yml`. For workflows that are not actively used all runs can be removed of course.